lib/views/customer_details.erb

```diff
@@ -22,6 +22,199 @@
       <div class="row">
         <div class="col-3"></div>
         <div class="col-6">
+        <% if @customer_details.nil? %>
+          <form class="pt-5" method="POST" action="/customer-details">
+            <div class="form-group">
+              <label for="customer-name">Customer Name<span class="text-danger">*</span></label>
+              <input
+                type="text"
+                class="form-control"
+                id="customer-name"
+                name="customer_name"
+              />
+              <%if @errors.include?(:missing_customer_name)%>
+                <p class="text-danger">Please enter a customer name</p>
+              <%end%>
+            </div>
+            <h3 class="pt-1 pb-1"> Shipping Details </h3>
+            <div class="form-group">
+              <label for="shipping-address-line1">Address Line 1<span class="text-danger">*</span></label>
+              <input
+                type="text"
+                class="form-control"
+                id="shipping-address-line1"
+                name="shipping_address_line1"
+              />
+              <%if @errors.include?(:missing_shipping_address_line1)%>
+                <p class="text-danger">Please enter a shipping address</p>
+              <%end%>
+            </div>
+            <div class="form-group">
+              <label for="shipping-address-line2">Address Line 2</label>
+              <input
+                type="text"
+                class="form-control"
+                id="shipping-address-line2"
+                name="shipping_address_line2"
+              />
+            </div>
+            <div class="form-group">
+              <label for="shipping-city">City<span class="text-danger">*</span></label>
+              <input
+                type="text"
+                class="form-control"
+                id="shipping-city"
+                name="shipping_city"
+              />
+              <%if @errors.include?(:missing_shipping_city)%>
+                <p class="text-danger">Please enter a shipping city</p>
+              <%end%>
+            </div>
+            <div class="form-group">
+              <label for="shipping-county">County<span class="text-danger">*</span></label>
+              <input
+                type="text"
+                class="form-control"
+                id="shipping-county"
+                name="shipping_county"
+              />
+              <%if @errors.include?(:missing_shipping_county)%>
+                <p class="text-danger">Please enter a shipping county</p>
+              <%end%>
+            </div>
+            <div class="form-group">
+              <label for="shipping-postcode">Postcode<span class="text-danger">*</span></label>
+              <input
+                type="text"
+                class="form-control"
+                id="shipping-postcode"
+                name="shipping_postcode"
+              />
+              <%if @errors.include?(:missing_shipping_postcode)%>
```

Comment:
Can guard clauses been used in erb files?

Reply:
They need to be if/else because we only want one error at a time to show

Reply:
I dont think you can do gaurd class but you can do single line code

```<%= content_tag(:p, class:'text-danger', data: 'Please enter a customer name') if @errors.include?(:missing_customer_name)%>```